services/QuillLMS/client/app/bundles/Teacher/components/dashboard/handy_actions.tsx

```diff
@@ -25,18 +26,22 @@ const ExploreDemoAction = ({setShowDemoModal}) => (
   </button>
 )
 
-const HandyActions = ({ linkedToClever, setShowDemoModal}) => (
-  <section className="handy-actions">
+const HandyActions = ({ linkedToClever, setShowDemoModal}) => {
+  const [errors, setErrors] = React.useState<string[]>([])
+  const [hasAppSetting, setHasAppSetting] = React.useState<boolean>(false);
+  if (!hasAppSetting) { handleHasAppSetting({appSettingSetter: setHasAppSetting, errorSetter: setErrors, key: 'demo', }) }
```

Comment:
This should go in a `React.useEffect` hook, with the second argument as an empty array - that's effectively the functional equivalent of the old `componentDidMount` method. That will have it only run once. Right now it's going to run every time the component rerenders, which shouldn't be a huge deal in your use case but isn't a pattern we should be following.